linkml/generators/fhirgen.py

```diff
@@ -0,0 +1,388 @@
+   - Nested objects: e.g. FHIR's `identifier`
+   - Mutation dict of func/lambda's
+   - Do all of the 1..x cardinalities in FHIR first
+todo later's
+ 1. XML output
+"""
+import csv
+import json
+import os
+from copy import deepcopy
+# from enum import Enum
+from pathlib import Path
+from typing import Callable, Dict, Iterator, TextIO, Tuple, Union
+
+import click
+
+from linkml_runtime.dumpers import yaml_dumper
+from linkml_runtime.linkml_model.meta import ClassDefinition, ClassDefinitionName, Definition, DefinitionName, \
+    Element, EnumDefinition, SchemaDefinition, SlotDefinition, SubsetDefinition, TypeDefinition
+from linkml_runtime.utils.formatutils import camelcase, underscore
+from linkml_runtime.utils.schemaview import SchemaView
+# from linkml.utils.generator import Generator, shared_arguments
+# todo: Do I need _ensure_ranked? If so, move out of docgen?
+from linkml.generators.docgen import _ensure_ranked
+from linkml.utils.generator import Generator
+
+
+# todo: make some of these class properties?
+FHIRGEN_DIR = os.path.join(os.path.dirname(__file__), 'fhirgen')
+MODEL_MAPPING_PATH = os.path.join(FHIRGEN_DIR, 'mapping__enum_valueset.csv')
+FHIR_VALUESET_FIELD_MUTATIONS: Dict[str, Callable] = {
+    'status': lambda x: 'unknown'  # todo: temp
+}
+
+
+class FhirValueSetGenerator(Generator):
+    """Generate FHIR ValueSet JSON from a LinkML YAML Enum schema"""
+    generatorname = os.path.basename(__file__)
+    generatorversion = "0.0.1"
+    visit_all_class_slots = False
+    genmeta = False  # todo: Do I need this?
+
+    def __init__(
+        self,
+        schema: Union[str, TextIO, SchemaDefinition],
+        outpath: str = None,
+        genmeta: bool = False,
+    ) -> None:
+        """
+        Creates a generator object that can write output
+
+        :param schema: path to schema file or schema object
+        :param outpath: path to save file or file(s)
+        :param genmeta:
+        """
+        self.sourcefile = schema
+        self.schemaview = SchemaView(schema)
+        self.schema = self.schemaview.schema
+        self.format = 'json'  # todo: support w/ CLI
+        self.outpath = outpath
+        self.genmeta = genmeta
+
+    def serialize(self, outpath: str = None) -> None:
+        """
+        Serialize Enums in schema to JSON
+
+        :param directory: relative or absolute path to directory in which documents are to be written
+        :param outpath: path to save file or file(s)
+        :return:
+        """
+        sv = self.schemaview
+        enums: Dict[str, EnumDefinition] = sv.all_enums()
+        # todo: `en` and `n` below are redundant?
+        for en, e in enums.items():
+            n = self.name(e)
+            d: Dict = self.serialize_enum(n, e)
+            out_str: str = json.dumps(d)
+            if outpath:
+                self._write(out_str, outpath, n)
+
+    @staticmethod
+    def model_mappings(path: str = MODEL_MAPPING_PATH) -> Dict[str, str]:
```

Comment:
Would like to use `pandas`, but it's not in the project requirements, so I used the standard library instead.

Reply:
Yes, pandas is a heavyweight package hence while developing `linkml` the team made a decision not to rely on it and use lightweight alternatives.